src/homeWork3/home_work3.ts

```diff
@@ -0,0 +1,206 @@
+
+//Так само у неї існують методи для обчислення балансу виходячи з поточного бюджету, додавання нових співробітників,
+
+//який враховує зміни балансу і перетворення з Попередньо найнятого на Співробітника або видалення Співробітника з
+
+//минулого відділу.
+
+//Сутність Попередньо найнятого співробітника має ім'я, прізвище, зарплата та номер банківського рахунку.
+
+//Сутність Співробітника - ім'я, прізвище, платіжну інформацію, зарплату, статус (активний, неактивний, у неоплачуваній відпустці) і знання про департамент,
+
+//до якого він прикріплений.
+
+//Так само у нас є сутність Бухгалтерія, яка є департаментом і має властивість баланс, а також методи
+
+//для взяття на баланс співробітника або департаменту, зняття з балансу і виплати зарплати для всього персоналу.
+
+//Попередньо найняті співробітники отримують зарплату за допомогою зовнішніх оплат, Співробітники (тільки активні) - за допомогою внутрішніх.
+
+
+// Енуми
+enum PaymentMethod {
+  Internal = 'internal',
+  External = 'external',
+}
+
+enum EmployeeStatus {
+  Active = 'active',
+  Inactive = 'inactive',
+  UnpaidLeave = 'unpaidLeave',
+}
+
+// Тіпи
+type Personal = {
+  name: string;
+  surname: string;
+};
+
+type Budgets = {
+  debit: number;
+  credit: number;
+};
+
+// Класи
+class PreHiredEmployee {
+  //Клас попередньо найнятий співробітник
+  name: Personal;
+  surname: Personal;
+  salary: number;
+  bankAccountNumber: string;
+
+  constructor(name: Personal, surname: Personal, salary: number, bankAccountNumber: string) {
+    this.name = name;
+    this.surname = surname;
+    this.salary = salary;
+    this.bankAccountNumber = bankAccountNumber;
+  }
+}
+
+class Employee {
+  //Клас співробітник
+  name: Personal;
+  surname: Personal;
+  paymentMethod: PaymentMethod;
+  salary: number;
+  status: EmployeeStatus;
+  department: Departments;
+
+  constructor(
+    name: Personal,
+    surname: Personal,
+    paymentMethod: PaymentMethod,
+    salary: number,
+    status: EmployeeStatus,
+    department: Departments
+  ) {
+    this.name = name;
+    this.surname = surname;
+    this.paymentMethod = paymentMethod;
+    this.salary = salary;
+    this.status = status;
+    this.department = department;
+  }
+}
+
+class Departments {
+  //Клас департамент
+  name: string;
+  domainName: string;
+  employees: Employee[];
+  budget: Budgets;
+
+  constructor(name: string, domainName: string, employees: Employee[], budget: Budgets) {
+    this.name = name;
+    this.domainName = domainName;
+    this.employees = employees;
+    this.budget = budget;
+  }
+
+  calculateBalance(): number {
+    //розрахунок балансу
+    return this.budget.debit - this.budget.credit;
+  }
+
+  addEmployees(newEmployees: Employee[]): void {
+    //додавання співробітника
+    this.employees.push(...newEmployees);
+
+    const costPerEmployee = 5000; // додаємо баланс нового співробітника
```

Comment:
I like that functionality. It feels like a real case. Cool. In that case, let's add that as a second function parameter with a default value of 5k. What do you think?

Reply:
ok, thank`s.